src/fmu/dataio/_metadata.py

```diff
@@ -37,12 +32,14 @@
 logger: Final = null_logger(__name__)
 
 
-def generate_meta_tracklog() -> list[meta.TracklogEvent]:
+def generate_meta_tracklog(
+    event: Literal["created", "merged"] = "created",
+) -> list[meta.TracklogEvent]:
```

Comment:
A bit unsure if it is two or three types. In the `meta.TracklogEvent` this field is typed as a `str` and the examples states three types. I can't see any reference to this `updated` event inside `dataio` so I'm quite certain dataio don't create metadata where `tracklog.event = "updated"`.

https://github.com/equinor/fmu-dataio/blob/0c2d04016875c2194b0f1777c37bb743e5880e1b/src/fmu/dataio/datastructure/meta/meta.py#L327-L329

Probably we should constrain the type for the event field inside the `meta.TracklogEvent` to an `Enum` or at least a `Literal` instead of `str`

Reply:
Created #669